File: Codesignal/Eric/71.MinimalNumberOfCoins.py
def solution(coins, price):
    result = 0
    for coin in coins[::-1]:
        result += (price // coin)
        price %= coin
    return result

def solution2(coins, price):
    res = 0
    for coin in reversed(coins):
        t, price = divmod(price, coin)
        res += t
    return res


print(solution([1,5,10,100], 239))

File: Codesignal/Eric/71.MinimalNumberOfCoins.js
function solution(coins, price) {
    let result = 0;
    coins.sort((a, b) => b-a).map(x => {
            result += ~~(price / x)
            price %= x
        })
    return result;
}

console.log(solution([1,5,10,100], 239))
